src/core_deku/state.ml

```diff
@@ -101,113 +135,176 @@ let rec apply_user_operation ?sender t operation_hash user_operation =
+                         {
+                           additional_operations = receipts;
+                           remaining_tickets =
+                             Contract_vm.Contract.tickets_mapping contract
+                             |> List.map ~f:fst;
+                           new_storage =
+                             Contract_vm.Contract.raw_storage contract;
+                         };
+                   });
+          })
+    |> Result.map_error ~f:(fun _ ->
+           `Invocation_error ("execution error", sender))
+
+and apply_contract_ops ~source ~sender (t, acc) x =
   match x with
   | Context.Operation.Invoke { tickets; destination; param } ->
     let%ok destination =
       Address.to_contract_hash destination
-      |> Result.of_option ~error:(`Invocation_error "invalid address") in
+      |> Result.of_option
+           ~error:
+             (`Invalid_address_error ("invalid address", sender, destination))
+    in
     let%ok payload =
       Contract_vm.Invocation_payload.of_bytes ~arg:param
-      |> Result.map_error ~f:(fun x -> `Invocation_error x) in
+      |> Result.map_error ~f:(fun x ->
+             `Invalid_payload (Format.sprintf "invalid_payload %s\n" x)) in
     let operation =
       User_operation.Contract_invocation
         { to_invoke = destination; argument = payload; tickets } in
     let operation = User_operation.make ~source operation in
-    apply_user_operation ~sender t operation.hash operation
+    let%ok { new_state; receipt } =
+      apply_user_operation ~sender t operation.hash operation in
+    Ok
+      ( new_state,
+        { user_operation = operation; operation_receipt = receipt } :: acc )
   | Transfer { ticket; amount; destination } ->
     let%ok destination =
       Address.to_key_hash destination
-      |> Result.of_option ~error:(`Invocation_error "invalid address") in
+      |> Result.of_option
+           ~error:
+             (`Invalid_address_error
+               ("invalid implicit account address", sender, destination)) in
     let operation = User_operation.Transaction { destination; amount; ticket } in
     let operation = User_operation.make ~source operation in
-    apply_user_operation ~sender t operation.hash operation
+    let%ok { new_state; receipt } =
+      apply_user_operation ~sender t operation.hash operation in
+    Ok
+      ( new_state,
+        { user_operation = operation; operation_receipt = receipt } :: acc )
 
 let apply_user_operation t hash user_operation =
   let report_error error =
     let message =
       match error with
-      | `Origination_error msg -> "Origination error: " ^ msg
-      | `Invocation_error msg -> "Invocation error: " ^ msg
-      | `Insufficient_funds -> "Insufficient funds" in
+      | `Origination_error (msg, _) -> "Origination error: " ^ msg
+      | `Invocation_error (msg, _) -> "Invocation error: " ^ msg
+      | `Insufficient_funds -> "Insufficient funds"
+      | `Invalid_payload _ -> "Invalid contract payload"
```

Comment:
Please, don't ignore error messages and log it back.

Reply:
Wdym?